### ISSUE TITLE
Named enum for string types

### ISSUE THREAD

**Issue by java2kus** [bcherny/json-schema-to-typescript#60]
How do I generate named enums for string types.   My JSON Schema looks like below:

```
"status": {
      "enum": [
        "I",
        "A",
        "C"
      ],
      "tsEnumNames": [
        "InProgress",
        "ForApproval",
        "Complete"
      ]
    },
```
Should generate the following Typescript 
```
  status: Status
....
export const enum Status {
   InProgress = "I"
   ForApproval = "A"
   Complete = "C"
}
```
Instead of the above, its generating  the flags without quotes that results in TypeScript error.   Can you please suggest, how can I generate the above typescript code?

**Comment by java2kus** [comment]
@bcherny Thanks a lot for clarifying.   Coming from a Java programming background, I am trying to wrap my head around Typescript :-).   I guess I cannot solve my specific problem of substitution.   But how I can achieve type safety on passing functions for the above.   

e.g.  I am trying to create a function where one of the parameters is status.   Currently I am defining this separately by adding a type like this which is leading to a lot of boilerplate and repetition in my code. In my case MyInterface has many more properties like taskDescription etc.

If the generated interface above generates types for String literals my problem will be solved.   Can you please let me know if this can be achieved in any way?

```typescript
  "taskDescription": {
      "type": "string"
  }
  "status": {
          "enum": ["InProgress", "ForApproval", "Complete"]
   }
==>
   export type Status = "InProgress" | "ForApproval" | "Complete";

    export interface MyInterface {
       taskDescription: string;
       status: Status;
    
   }
.....
    generateEvent(status: Status, taskDescription: string) {  ... }
```

**Comment by bcherny** [comment]
@java2kus I'm not sure I understand what you're trying to do. The code you gave above will compile perfectly, is there something more you're trying to generate?

**Comment by java2kus** [comment]
@bcherny I was checking if the above code can be generated using json-schema-to-typescript library.   Right now after the code generation, I am manually making all the above changes in my code.

**Comment by bcherny** [comment]
There is no support for named union types today.

```json
"taskDescription": {
      "type": "string"
  }
  "status": {
          "enum": ["InProgress", "ForApproval", "Complete"]
   }
```

Will generate 

```ts
export interface MyInterface {
  taskDescription: string;
  status: "InProgress" | "ForApproval" | "Complete";
}
```

**Comment by java2kus** [comment]
@bcherny This is how I have manually coded the string based enums in Typescript.   I thought it would be useful for any future code generation enhancements.   I am using Typescript 2.2.

```typescript
export enum Status {
  InProgress = <any> 'I',
  ForApproval = <any> 'A',
  Complete = <any> 'C'
}

```

**Comment by bcherny** [comment]
@java2kus I don't think that actually works. It tricks the compiler into compiling the enum, but you don't get any type safety with it. What you want is https://github.com/Microsoft/TypeScript/issues/1206, or restructuring your code to use a number enum/union of string literal types.

```ts
export enum Status {
  InProgress = <any> 'I',
  ForApproval = <any> 'A',
  Complete = <any> 'C'
}

Status.A // Error

if (Status.InProgress === 'I') { } // Error
```

**Comment by pasawant** [comment]
@bcherny i too wanted to generate string enum from schema:

https://www.typescriptlang.org/docs/handbook/enums.html#string-enums
